crates/autopilot/src/domain/settlement/coded/tokenized.rs

```diff
@@ -0,0 +1,26 @@
+use ethcontract::{Address, Bytes, U256};
+
+// Original type for input of `GPv2Settlement.settle` function.
+pub type Settlement = (
+    Vec<Token>,
+    Vec<ClearingPrice>,
+    Vec<Trade>,
+    [Vec<Interaction>; 3],
+);
+
+pub type Token = Address;
+pub type ClearingPrice = U256;
+pub type Trade = (
+    U256,            // sellTokenIndex
+    U256,            // buyTokenIndex
+    Address,         // receiver
+    U256,            // sellAmount
+    U256,            // buyAmount
+    u32,             // validTo
+    Bytes<[u8; 32]>, // appData
+    U256,            // feeAmount
+    U256,            // flags
+    U256,            // executedAmount
+    Bytes<Vec<u8>>,  // signature
+);
+pub type Interaction = (Address, U256, Bytes<Vec<u8>>);
```

Comment:
I don't think these should be public (they are only used privately by `Settlement` to construct itself. Can we either inline them there or otherwise restrict visibility to not expose those details to the entire codebase?

Reply:
updated to pub(super)